okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/DistinguishedNameParser.java

```diff
@@ -0,0 +1,410 @@
+/*
```

Comment:
I think @ejona86 has mentioned it or am I misunderstanding your question?
```
There are much worse "internal" classes we are using. The reason to copy it is so that we can upgrade it on our schedule instead of restricting the versions of okhttp our users can use.
```